js/views/modals/Settings/Page.js

```diff
@@ -30,9 +30,54 @@ export default class extends baseVw {
       'click .js-avatarRight': 'avatarRightClick',
       'click .js-headerLeft': 'headerLeftClick',
       'click .js-headerRight': 'headerRightClick',
+      'change .js-colorPicker': 'handleColorChosen',
+      'change .js-colorCode': 'handleColorCodeEntered',
     };
   }
 
+  /** Handles when a hex color code is entered by updating color picker. */
+  handleColorCodeEntered(event) {
+    const colorPickerId = $(event.target).data('color-picker-id');
+    const $colorPicker = this.getCachedElement(colorPickerId);
+    const newHexColorCode = event.target.value;
+
+    // If the text passes a basic RegExp for a valid 6 digit hex value,
+    // update the color picker's color.
+    if (/^#([0-9a-f]{6})$/i.test(newHexColorCode)) {
+      $colorPicker.val(newHexColorCode);
+    }
+  }
+
+  /** Handles when a color is chosen from the color picker by updating hex color code text. */
+  handleColorChosen(event) {
+    const hexInputId = $(event.target).data('hex-input-id');
+    const $hexInput = this.getCachedElement(hexInputId);
+    const newColor = event.target.value;
+
+    $hexInput.val(newColor);
+  }
+
+  /** Returns a given jquery Element or a locally cached version in this view for optimization. */
+  getCachedElement(elementId) {
```

Comment:
This isn't a bad we to abstract the caching of dom elements within a view. At some point, we may even want to consider integrating it into the base view.

A few thoughts:
- You need to clear the map in render() because at that point the DOM elements held in the cache are no longer in the DOM or associate with this view - the view has a fresh set of DOM elements and you don't want to get the old ones when calling this method. You could see how this happens by interacting with the color fields, clicking Save (which re-renders) and then finding the color field functionality implemented here no longer works because you're referencing cached elements that are no longer in this view.
- rather than semantically tie it to an id by using the name `elementId`, should we use something like `selector` and then you could pass in any selector, since often times we're really just working off of classes?